src/main/java/ch/njol/skript/expressions/ExprKickMessage.java

```diff
@@ -0,0 +1,78 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.lang.EventRestrictedSyntax;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerKickEvent;
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Kick Message")
+@Description("The kick message that is displayed on-screen when a player is kicked.")
+@Examples({
+	"on kick:",
+		"\tkick message is 'Invalid hotbar selection (Hacking?)'",
+		"\tcancel event"
+})
+@Since("INSERT VERSION")
+@Events("Kick")
+public class ExprKickMessage extends SimpleExpression<String> implements EventRestrictedSyntax {
+
+	static {
+		Skript.registerExpression(ExprKickMessage.class, String.class, ExpressionType.SIMPLE, "kick (message|reason)");
```

Comment:
```suggestion
		Skript.registerExpression(ExprKickMessage.class, String.class, ExpressionType.SIMPLE, "(display[ed]|on-screen) kick (message|reason)");
``` there should be a way to differentiate the chat kick message from this one